### ISSUE TITLE
Numerical discrepancy in torch.special.erfcx between CPU and CUDA implementations

### ISSUE THREAD

**Issue by beanduan22** [pytorch/pytorch#179785]
### 🐛 Describe the bug

```
import torch
import torch.nn as nn

torch.manual_seed(0)

fc1 = nn.Linear(8, 8)
bn  = nn.BatchNorm1d(8)
fc2 = nn.Linear(8, 8)
logsigmoid = nn.LogSigmoid()

def forward(x):
    x = fc1(x)
    x = torch.nn.functional.hardshrink(x)
    x = bn(x)
    x = fc2(x)
    x = torch.sqrt(x.abs())          # ensure non-negative for sqrt
    x = torch.special.erfcx(x)       # scaled complementary erf — CPU/GPU differ
    return logsigmoid(x)

fc1_g = nn.Linear(8, 8).cuda(); fc1_g.load_state_dict(fc1.state_dict())
bn_g  = nn.BatchNorm1d(8).cuda(); bn_g.load_state_dict(bn.state_dict())
fc2_g = nn.Linear(8, 8).cuda(); fc2_g.load_state_dict(fc2.state_dict())
ls_g  = nn.LogSigmoid().cuda()

def forward_gpu(x):
    x = fc1_g(x)
    x = torch.nn.functional.hardshrink(x)
    x = bn_g(x)
    x = fc2_g(x)
    x = torch.sqrt(x.abs())
    x = torch.special.erfcx(x)
    return ls_g(x)

x = torch.randn(4, 8)
cpu_out = forward(x)
gpu_out = forward_gpu(x.cuda()).cpu()

diff = (cpu_out - gpu_out).abs()
print(f"max |cpu - gpu|: {diff.max():.4e}")

# Minimal 2-line version:
print("\nMinimal:")
v = torch.rand(4, 8) * 3          # erfcx is most sensitive in [0, 3]
print(f"erfcx cpu/gpu diff: {(torch.special.erfcx(v) - torch.special.erfcx(v.cuda()).cpu()).abs().max():.4e}")
```

### Versions

  2.9.1+cu128 (PyTorch 2.9.1, CUDA 12.8)   

torch.special.erfcx produces noticeably different results between CPU and CUDA for the same inputs.
The discrepancy can be amplified in typical neural network pipelines and may affect numerical stability and reproducibility.

cc @ptrblck @msaroufim @eqy @jerryzh168 @tinglvv @nWEIdia @mruberry @kshitij12345

**Comment by malfet** [comment]
And your ask is? See https://github.com/pytorch/pytorch/wiki/Finding-Or-Reporting-Issues#reporting-accuracy-problems
